code/datums/components/reagent_refiller.dm

```diff
@@ -0,0 +1,70 @@
+/**
+ * ## Reagent refiller
+ * Refills any drinks poured out of the reagent container (and is allowed within the whitelisted reagents).
+ */
+/datum/component/reagent_refiller
+	///Time to refill
+	var/time_to_refill
+	///Cell to use power of
+	var/datum/weakref/cell_to_use
+	///Amount of power to use from the cell
+	var/power_to_draw
+	///Whitelist of reagents allowed to be synthesized
+	var/list/whitelisted_reagents
+
+/datum/component/reagent_refiller/Initialize(
+	time_to_refill = 60 SECONDS,
+	obj/item/stock_parts/cell/cell,
+	power_to_draw = 30,
+	whitelisted_reagents = list(/datum/reagent/consumable)
+)
+	if(!istype(parent, /obj/item/reagent_containers))
+		return COMPONENT_INCOMPATIBLE
+
+	src.time_to_refill = time_to_refill
+	if (!isnull(cell))
+		src.cell_to_use = WEAKREF(cell)
+	src.power_to_draw = power_to_draw
+	src.whitelisted_reagents = whitelisted_reagents
+
+	return ..()
+
+/datum/component/reagent_refiller/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ITEM_AFTERATTACK, .proc/refill)
```

Comment:
This is kinda off, shouldn't this be some reagent container signal on reagent removal ?

Reply:
Probably, i was just using what the original mechanic used